tutorials/source-modeling/plot_beamformer_lcmv.py

```diff
@@ -0,0 +1,169 @@
+"""
+Source reconstruction using an LCMV beamformer
+==============================================
+
+This tutorial gives an overview of the beamformer method
+and shows how to use an LCMV beamformer to reconstruct source
+activity.
+
+.. contents:: Page contents
+   :local:
+   :depth: 2
+
+A beamformer is a spatial filter that reconstructs source activity by scanning
+through a grid of pre-defined source points and estimating activity at each of
+those source points independently.
+The beamforming method applied in this tutorial is the linearly constrained
+minimum variance (LCMV) beamformer [1]_ which operates on time series.
+Frequency-resolved data can be reconstructed with the dynamic imaging of
+coherent sources (DICS) beamforming method [2]_.
+"""
+# Author: Britta Westner <britta.wstnr@gmail.com>
+#
+# License: BSD (3-clause)
+
+# sphinx_gallery_thumbnail_number = 5
+
+import mne
+from mne.datasets import sample
+from mne.beamformer import make_lcmv, apply_lcmv
+
+###############################################################################
+# Data processing
+# ---------------
+# We will use the sample data set for this tutorial and reconstruct source
+# activity on the trials with left auditory stimulation. Note that
+# beamformers are usually computed in a :class:`volume source space
+# <mne.VolSourceEsitmate>`, as a visualization
+# of only the surface activation can misrepresent the data.
+
+data_path = sample.data_path()
+subjects_dir = data_path + '/subjects'
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+fwd_fname = data_path + '/MEG/sample/sample_audvis-meg-vol-7-fwd.fif'
+
+# Read forward model
+forward = mne.read_forward_solution(fwd_fname)
```

Comment:
I know in analysis scripts it's common to do all the file I/O early on, but I wonder if it might be better for the tutorial to wait to load the forward solution until later, when it is needed (and when the need for it is explained).  Something more like this:

- load the (raw) data, epoch it, and plot an evoked butterfly plot
- explain why we need covariances, compute them, plot them, interpret them
- explain that we also need a forward solution, and why we need it / what role it plays in the beamformer (the "why" part is currently missing I think). This is where I would load the forward.  I would also add a cross-reference to some other example, tutorial, or function explaining how to *compute* the forward model (as users will typically need to do that step)
- compute the spatial filter
- apply the spatial filter
- visualize reconstructed source

What do you think?

Reply:
Tried to incorporate this now, I think it does make it easier to follow!